end2end-test-examples/gcs/src/main/java/io/grpc/gcs/JavaClient.java

```diff
@@ -8,26 +8,29 @@
 import com.google.cloud.storage.BlobId;
 import com.google.cloud.storage.BlobInfo;
 import com.google.cloud.storage.Storage;
-import com.google.cloud.storage.StorageOptions;
+import com.google.common.io.ByteStreams;
 import java.io.IOException;
 import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
 import java.util.Random;
 import java.util.concurrent.Executors;
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 import java.util.logging.Logger;
 
-public class JavaClient {
+public abstract class JavaClient {
```

Comment:
Can we do this without having other variants like `JavaGrpcClient`, `JavaGrapcDirectPathClient`, and `JavaJsonClient`? This would also work but I'd like to make this consistent with others (e.g. `GcsioClient` here and C++ tests). For directpath, you can check `Args.dp` to see whether directpath is used or not. (Sorry about cryptic acronym).

Reply:
I've made this class concrete, and added logic to the constructor to configure the StorageOptions based upon Args provided.